doc/releases/changelog-dev.md

```diff
@@ -54,6 +54,9 @@
 * Ensure that `qml.data.load` returns datasets in a stable and expected order.
   [(#3856)](https://github.com/PennyLaneAI/pennylane/pull/3856)
 
+* Made `qml.OrbitalRotation` consistent with the interleaved Jordan-Wigner ordering.
```

Comment:
I'm wondering: Users might have been using `OrbitalRotation`, using its current behaviour, which is not entirely wrong, but just inconsistent regarding the qubit ordering. So maybe it might be worth to add a "breaking change" entry as well, even though this is a bug fix, technically speaking... :thinking: